src/backend/app/Services/ClusterRevenueService.php

```diff
@@ -82,7 +82,7 @@ static function ($q) {
      */
     public function getTransactionsForWeeklyPeriod(int $clusterId, array $period, ?int $connectionType = null): Collection {
         return $this->transaction->newQuery()
-            ->selectRaw('DATE_FORMAT(created_at,\'%Y-%m\') as period , SUM(amount) as revenue')
+            ->selectRaw('DATE_FORMAT(created_at,\'%Y-%m\') as period , SUM(amount) as revenue, WEEKOFYEAR(created_at) as week')
```

Comment:
I don't understand why this wasn't necessary before Larastan fixes 😂 But @alaincodes24 suggested the same change.

Reply:
The error seems to have been suppressed because when you convert the transaction object to an array, you get only the fields `[ period => '', revenue => ''`, indexing that array by "week" throws an error.

We starting converting the revenueTransaciton returned db object to an array after the Larastan fixes.